core/src/klio_core/config/_preprocessing.py

```diff
@@ -77,20 +76,12 @@ def _transform_io_list(io_subsection_list):
         }
         """
 
-        type_counters = collections.defaultdict(int)
         io_dict = {}
-        for conf in io_subsection_list:
+        for count, conf in enumerate(io_subsection_list):
             if "name" in conf:
                 name = conf["name"]
-                # TODO: right now "name" isn't supported in IOConfig (conflicts
-                # with existing "name" attribute), once that's fixed we
-                # shouldn't drop name here
-                conf.pop("name")
             else:
-                type_name = conf.get("type", "unknown")
-                type_id = type_counters[type_name]
-                type_counters[type_name] += 1
-                name = "{}{}".format(type_name, type_id)
+                name = count
```

Comment:
Is there a reason why the type is being dropped from the generated name?  I'm not strongly for keeping it but I also don't see any clear reason for removing it.

Reply:
I think i remember we discussed as a team and thought there was weirdness since there is another area https://github.com/spotify/klio/blob/develop/exec/src/klio_exec/commands/run.py#L466 where we again append an index to the type. i THINK it's used because multiple event reading requires a unique `label` (I'll cite this when i find the link) and this generated name was for the unique label. I'll ask the team and confirm if we want to keep or not and I can pluck off that commit if we do indeed want to keep it.

Theres also this TODO written by @econchick that touches on stuff disappearing if we convert to enforcing a dictionary for event inputs and i think the two are related. https://github.com/spotify/klio/blob/develop/exec/src/klio_exec/commands/run.py#L460